### ISSUE TITLE
[nin10kitgui] Segmentation Fault: 11 | MacOS 

### ISSUE THREAD

**Issue by Shubhang** [TricksterGuy/nin10kit#37]
Running nin10gui on macOS v.11.2.3 returns "Segmentation Fault:11" This happens with both versions: homebrew and compiled from source.

As we talked before, I tried to compile it from source using the instructions in the readme. But I don't know what flags to use to compile it with debugging information. 

Running nin10kitgui with GDB throws a message that states "No debugging symbols found in nin10kitgui"

**Comment by TricksterGuy** [comment]
I'm not sure I've ever had a conversation with you about this?

I was going to redirect you to [skyman's homebrew repo](https://github.com/skyman/homebrew-gt), however it seems he's not actively maintaining it anymore.

If you can compile the program yourself using cmake and clang, you should be able to get debug symbols by setting CMAKE_BUILD_TYPE to Debug and compiling the program after that.

All I want is the backtrace when the segfault occurs to start with, if I can't get that information then I won't be able to resolve this.

**Comment by TricksterGuy** [comment]
More specifically if you aren't using cmake-gui

the step

Generate a Makefile with cmake: `cmake ..`

You will need to add `-DCMAKE_BUILD_TYPE=Debug` to that command

`cmake -DCMAKE_BUILD_TYPE=Debug ..`

As I said if you don't give me a stacktrace then I'm not going to go out of my way right now to fix this.

**Comment by Shubhang** [comment]
Thanks! We had a conversation about this when I took CS2110 last semester :) 

I compiled the program using your instructions and ran it using using lldb to get a stacktrace.  

Here is the stacktrace: 

```
Process 5681 launched: '/usr/local/bin/nin10kitgui' (x86_64)
2021-06-18 08:35:14.372056-0400 nin10kitgui[5681:139014] SecTaskLoadEntitlements failed error=22 cs_flags=20, pid=5681
2021-06-18 08:35:14.372112-0400 nin10kitgui[5681:139014] SecTaskCopyDebugDescription: nin10kitgui[5681]/0#-1 LF=0
Process 5681 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=2, address=0x7ffeef3ffff8)
    frame #0: 0x00007fff202ba818 libsystem_malloc.dylib`tiny_malloc_should_clear + 228
libsystem_malloc.dylib`tiny_malloc_should_clear:
->  0x7fff202ba818 <+228>: callq  0x7fff202bad9f            ; tiny_malloc_from_free_list
    0x7fff202ba81d <+233>: movq   %rax, %r12
    0x7fff202ba820 <+236>: testq  %rax, %rax
    0x7fff202ba823 <+239>: jne    0x7fff202baaf0            ; <+956>
Target 0: (nin10kitgui) stopped.

```
______

One quick note, I had to remove the "-pg" flag from CMakeLists.txt (Line 19) to successfully run `make` and `sudo make install`

In the presence of "-pg", I was getting the following error
`clang: error: the clang compiler does not support -pg option on versions of OS X 10.9 and later
`

**Comment by Shubhang** [comment]
Update: So I had to run `brew upgrade` and `brew cleanup` for a separate project. 

Apparently that fixed this issue. Now running `nin10kitgui`does not result in a seg fault and the gui opens up just fine. 

I think we can close this issue!

____

In the output of brew upgrade some of the packages relevant to this project were updated including 

- cmake 3.20.3 -> 3.20.4

- gcc 11.1.0 -> 11.1.0_1

- image magick 7.0.11-14 -> 7.1.0-0 

- wxMac 3.0.5.1_1 -> 3.0.5.1_2
